src/Symfony/Component/Messenger/Transport/Receiver/ReceiverInterface.php

```diff
@@ -21,11 +23,40 @@ interface ReceiverInterface
     /**
      * Receive some messages to the given handler.
      *
-     * The handler will have, as argument, the received {@link \Symfony\Component\Messenger\Envelope} containing the message.
+     * The handler will have, as argument, the received {@link \Symfony\Component\Messenger\Envelope}.
+     * The reveiver should add a RetryCountStamp to track the current retry number of this message.
      * Note that this envelope can be `null` if the timeout to receive something has expired.
+     *
+     * If the received message cannot be decoded, the message should
+     * be removed from the queue and the MessageDecodingFailedException be thrown
+     *
+     * @throws TransportException If there is an issue communicating with the transport
      */
     public function receive(callable $handler): void;
 
+    /**
+     * Acknowledge that the current message was handled (i.e. remove from queue).
+     *
+     * @throws TransportException If there is an issue communicating with the transport
+     */
+    public function acknowledgeCurrentMessage(): void;
+
+    /**
+     * Called when the handling of the current message has failed.
+     *
+     * @throws TransportException If there is an issue communicating with the transport
+     */
+    public function rejectCurrentMessage(): void;
+
+    /**
+     * Called when the handling of the current message failed but should be retried.
+     *
+     * @param int The delay to apply in milliseconds
+     *
+     * @throws TransportException If there is an issue communicating with the transport
+     */
+    public function retryCurrentMessage(int $delay): void;
```

Comment:
These all rely on being called from inside the `receive()` loop, because that keeps temporary "state" about which message is currently being handled. That's a key change here, which simplifies a lot, but which I want to make sure won't cause issues.